tests/pkg_openwsn/main.c

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       OpenWSN test application
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "od.h"
+#include "fmt.h"
+#include "shell.h"
+#include "openwsn.h"
+#include "net/ipv6/addr.h"
+
+#include "opendefs.h"
+#include "scheduler.h"
+#include "02a-MAClow/IEEE802154E.h"
+#include "02b-MAChigh/neighbors.h"
+#include "03b-IPv6/icmpv6rpl.h"
+#include "04-TRAN/openudp.h"
+#include "cjoin.h"
+#include "cross-layers/openqueue.h"
+#include "cross-layers/idmanager.h"
+#include "cross-layers/packetfunctions.h"
+
+extern idmanager_vars_t idmanager_vars;
+extern icmpv6rpl_vars_t icmpv6rpl_vars;
+extern neighbors_vars_t neighbors_vars;
+extern openqueue_vars_t openqueue_vars;
+extern schedule_vars_t schedule_vars;
+extern scheduler_dbg_t scheduler_dbg;
+
+udp_resource_desc_t uinject_vars;
+char addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+void uinject_sendDone(OpenQueueEntry_t *msg, owerror_t error)
+{
+    (void)error;
+
+    printf("msg.l2_sendDoneError: %x\n", msg->l2_sendDoneError);
+    openqueue_freePacketBuffer(msg);
+    puts("Send success");
+}
+
+void uinject_receive(OpenQueueEntry_t *pkt)
+{
+    printf("Received %i bytes on port %i\n", (int)pkt->length,
+           pkt->l4_destination_port);
+    od_hex_dump(pkt->payload, pkt->length, OD_WIDTH_DEFAULT);
+    openqueue_freePacketBuffer(pkt);
+}
+
+void uinject_init(void)
+{
+    uinject_vars.port = WKP_UDP_INJECT;
+    uinject_vars.callbackReceive = &uinject_receive;
+    uinject_vars.callbackSendDone = &uinject_sendDone;
+    openudp_register(&uinject_vars);
+}
+
+char *_array_2_string(const uint8_t *addr, size_t addr_len, char *out)
+{
+    char *res = out;
+
+    assert((out != NULL) && ((addr != NULL) || (addr_len == 0U)));
+    out[0] = '\0';
+    for (size_t i = 0; i < addr_len; i++) {
+        out += fmt_byte_hex((out), *(addr++));
+        *(out++) = (i == (addr_len - 1)) ? '\0' : ':';
+    }
+    return res;
+}
+
+static int ifconfig_cmd(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    open_addr_t *hwaddr;
+    open_addr_t temp_my128bID;
+
+    memcpy(&temp_my128bID.addr_128b[0], &idmanager_vars.myPrefix.prefix, 8);
+    memcpy(&temp_my128bID.addr_128b[8], &idmanager_vars.my64bID.addr_64b, 8);
+
+    ipv6_addr_to_str(addr_str, (ipv6_addr_t *)temp_my128bID.addr_128b,
+                     sizeof(addr_str));
+    printf("inet6 %s\n", addr_str);
+
+    hwaddr = idmanager_getMyID(ADDR_16B);
+    printf("hwaddr short: %s", _array_2_string(hwaddr->addr_16b, 2, addr_str));
```

Comment:
Maybe add a `\t` at the beginning of the output. Here and below as well. I think it would make the output of ifconfig a bit nicer.

Reply:
This is how it would look:

```
2020-06-30 09:31:57,069 # inet6 bbbb::9635:9a92:4e3d:6578
2020-06-30 09:31:57,074 #       hwaddr short: 07:C2         long: 96:35:9A:92:4E:3D:65:78
2020-06-30 09:31:57,075 #       panid: CA:FE
2020-06-30 09:31:57,075 #
2020-06-30 09:31:57,077 #       IEEE802154E sync: 1
2020-06-30 09:31:57,077 #
2020-06-30 09:31:57,079 #       6tsch joined: 1
2020-06-30 09:31:57,080 #
2020-06-30 09:31:57,082 #       RPL rank: 2816
2020-06-30 09:31:57,085 #       RPL parent: 82:6B:DE:EC:58:34:65:78
2020-06-30 09:31:57,089 #       RPL DODAG ID: bbbb::826b:deec:5834:6578
```

Reply:
if could have something comparable to ifconfig that woud be great:
```
> ifconfig
ifconfig
Iface  6  HWaddr: 52:8F:11:C9:A8:9B 
          L2-PDU:1500 MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
          Link type: wired
          inet6 addr: fe80::508f:11ff:fec9:a89b  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ffc9:a89b
          
          Statistics for Layer 2
            RX packets 34  bytes 4632
            TX packets 8 (Multicast: 1)  bytes 1248
            TX succeeded 8 errors 0
          Statistics for IPv6
            RX packets 34  bytes 4156
            TX packets 8 (Multicast: 1)  bytes 1136
            TX succeeded 8 errors 0
> 
```

Do you have access to the iface number with openwsn ?

Reply:
Nope. `gnrc_netif` is not used.

Reply:
So here is how it looks like:

```
2020-06-30 12:39:35,078 # Iface  3      HWaddr: 07:C2  NID: CA:FE
2020-06-30 12:39:35,078 #
2020-06-30 12:39:35,081 #               Long HWaddr: 96:35:9A:92:4E:3D:65:78
2020-06-30 12:39:35,085 #               inet6 addr: bbbb::9635:9a92:4e3d:6578
2020-06-30 12:39:35,085 #
2020-06-30 12:39:35,088 #               IEEE802154E sync: 1
2020-06-30 12:39:35,090 #               6TiSCH joined: 1
2020-06-30 12:39:35,090 #
2020-06-30 12:39:35,092 #               RPL rank: 523
2020-06-30 12:39:35,095 #               RPL parent: 82:6B:DE:EC:58:34:65:78
2020-06-30 12:39:35,096 #               RPL children:
2020-06-30 12:39:35,099 #                       96:8D:13:13:55:32:7E:78
2020-06-30 12:39:35,103 #               RPL DODAG ID: bbbb::826b:deec:5834:6578

```